### ISSUE TITLE
Default seed makes a random forest generate same estimators

### ISSUE THREAD

**Issue by zemlyansky** [mljs/random-forest#17]
Hi! Thanks for bringing random forests to the JS world! I was experimenting with making _ml-random-forest_ produce feature importance values and found this strange behavior. It can be reproduced with  Iris or any other training data:

```javascript
var IrisDataset = require('ml-dataset-iris')
var { RandomForestClassifier } = require('ml-random-forest')
var trainingSet = IrisDataset.getNumbers()
var predictions = IrisDataset.getClasses().map((elem) => IrisDataset.getDistinctClasses().indexOf(elem))

var options = {
  maxFeatures: 0.8,
  replacement: true,
  nEstimators: 25
}

var classifier = new RandomForestClassifier(options)
classifier.train(trainingSet, predictions)                                                                                                    
console.log(classifier.indexes)
console.log(JSON.stringify(classifier.estimators[0]) === JSON.stringify(classifier.estimators[1]))
```
Produces:
```
[
  [ 2, 3, 0 ], [ 2, 3, 0 ], [ 2, 3, 0 ],
  [ 2, 3, 0 ], [ 2, 3, 0 ], [ 2, 3, 0 ],
  [ 2, 3, 0 ], [ 2, 3, 0 ], [ 2, 3, 0 ],
  [ 2, 3, 0 ], [ 2, 3, 0 ], [ 2, 3, 0 ],
  [ 2, 3, 0 ], [ 2, 3, 0 ], [ 2, 3, 0 ],
  [ 2, 3, 0 ], [ 2, 3, 0 ], [ 2, 3, 0 ],
  [ 2, 3, 0 ], [ 2, 3, 0 ], [ 2, 3, 0 ],
  [ 2, 3, 0 ], [ 2, 3, 0 ], [ 2, 3, 0 ],
  [ 2, 3, 0 ]
]
true
```
So without manually resetting the seed with `{seed: undefined}` a random forest is not better than a decision tree. 

That show two issues:
1. Having default seed is not expected behavior. Random forest results should not be deterministic.
2. When the seed is set, it should not prevent splitting data on different chunks for training. Maybe generate new seeds for each sample with `seed+1`, `seed+2` or something else?

I can create a PR to fix this

**Comment by tarixgit** [comment]
Also just recognised.

**Comment by aiday-mar** [comment]
closed by 5ca902d074257b82be5947fc56478ab6d369f7b0

**Comment by aiday-mar** [comment]
closed by 5ca902d074257b82be5947fc56478ab6d369f7b0
